code/datums/datum.dm

```diff
@@ -9,6 +10,7 @@
 	var/datum/weakref/weak_reference
 	var/hidden_from_codex = FALSE //set to TRUE if you want something to be hidden.
 
+

```

Comment:
You're missing all the code that cleans up components

Reply:
Added